arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+# Bucket management in soda-multicloud
+
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10), [Vineela Pachchipulusu]()
+
+
+## Motivation and background
+
+In the cloud world, every cloud vendor provides storage space to users. Generally in terms of bucket and folder. A user can store the data/objects in a specific bucket or folder.
+
+SODA multi-cloud(gelato) is a project which manages multiple cloud vendors from one place. And also interfaces to use cloud services. So it is important for soda to provide bucket management support for various cloud vendors. So that user can create and manage buckets from one place(soda multi-cloud).
+
+## Goals
+The document provides the use case, design and implementation guidelines for supporting bucket management from SODA platform. It includes creation, deletion, object uploading, migration etc. in various cloud providers.
```

Comment:
each vendor supports different configurations on security, versioning etc. How is that unified here ? in SODA ?

Reply:
Will the model for new Bucket management be generic enough to provide the common functionalities/features like Lifecycle rules, SSE, etc. across multiple clouds?

Reply:
Yes. model for new bucket management is generic to provide common functionalities across different cloud vendors.